api/grpc/mpi/v1/command.proto

```diff
@@ -0,0 +1,358 @@
+message Exporter {
+    // how often to report in google.protobuf.Timestamp format
+    google.protobuf.Timestamp report_interval = 1;
+    // connection information to send data to a particular destination
+    ConnectionSettings connection_settings = 2;
+}
+
+// A set of connection information and it's associated auth, tls and backoff configurations
+message ConnectionSettings {
+    // Server settings that include connection information
+    Server server = 1;
+    // Authentication settings
+    Auth auth = 2;
+    // Optional TLS settings
+    TLSSetting tls = 3;
+    // backoff settings associated with this connection
+    Backoff backoff = 4;
+}
+
+// Server settings like hostname
+message Server {
+    // the host information
+    string host = 1;
+    // the port information
+    int32 port = 2;
+    // Connection Type enumeration
+    enum ConnectionType {
+        // Default connection type
+        UNKNOWN = 0;
+        // HTTP connection type
+        HTTP = 1;
+        // gRPC connection type
+        GRPC = 2;
+        // unix socket connection type
+        UNIX = 3;
+    }
+    // Specific connection type
+    ConnectionType connection_type = 3;
+}
+
+// Authentication settings
+message Auth {
+    // A token
+    string token = 1;
+}
+
+message TLSSetting {
+    // enable tls
+    bool enable      = 1;
+    // tls cert
+    string cert      = 2;
+    // tls key
+    string key       = 3;
+    // certificate authoirty cert
+    string ca        = 4;
+    // enable verification of a server's certificate chain and host name
+    bool skip_verify = 5;
+}
+
+// backoff settings
+message Backoff {
+    // First backoff time interval in seconds
+    int64 initial_interval = 1;
+    // Random value used to create range around next backoff interval
+    double randomization_factor = 2;
+    // Value to be multiplied with current backoff interval
+    double multiplier = 3;
+    // Max interval in seconds between two retries
+    int64 max_interval = 4;
+    // Elapsed time in seconds after which backoff stops.
+    // It never stops if max_elapsed_time == 0.
+    int64 max_elapsed_time = 5;
+}
+
+// A set of runtime NGINX configuration that gets populated 
+message NGINXConfig {
+    // where the binary location is, if empty, this is a remote instance
+    string binary_path = 1;
+    // the hostname associated with NGINX
+    string hostname = 2;
+    // the ip address associated with NGINX
+    string ip_address = 3;
+    // the stub status API location
+    string stub_status = 4;
+    // a list of access_logs
+    repeated string access_logs = 5;
+    // a list of error_logs
+    repeated string error_logs = 6;
+}
+
+// A set of runtime NGINX configuration that gets populated 
+message NGINXPlusConfig {
+    // where the binary location is, if empty, this is a remote instance
+    string binary_path = 1;
+    // the hostname associated with NGINX Plus
+    string hostname = 2;
+    // the ip address associated with NGINX Plus
```

Comment:
nit, not sure what "associated" means, clarify?

Reply:
this payload is likely to change during implementation. I added it for reference but don't understand it's purpose. It was mentioned in the spec so I put it here for illustration. I can remove it and we can build the proper payload as we go